test/jdk/javax/swing/text/BoxView/BaselineTest.java

```diff
@@ -0,0 +1,182 @@
+import javax.swing.text.View;
+import javax.swing.text.ViewFactory;
+
+public class BaselineTest {
+
+    static final String INSTRUCTIONS = """
+        Test that components displayed in a JTextPane properly respect their vertical alignment.
+        There are two text panes, stacked vertically with similar content except the bottom components are taller.
+        The content consists of a leading and trailing text string, with pink coloured components between.
+        The text string content means the strings 'Default Size Text' and 'Large Size Text'.
+        Text content baseline is at the bottom of CAPITAL letters in the text.
+        Each pink component has a string displaying its alignment setting in the range 0.0 to 1.0
+        NB: The position of the strings "align = 0.0" etc is not important, it is the component position that matters.
+        0.0 means it should be aligned with its top at the text content baseline,
+        1.0 means it should be aligned with its bottom at the text content baseline.
+        A value in between will be a proportional alignment, eg 0.5 is centered on the text content baseline
+        If the content displays as described, the test PASSES.
+    """;
+
+    public static void main(String[] args) throws Exception {
+       PassFailJFrame.builder()
+            .title("BaselineTest Test Instructions")
+            .instructions(INSTRUCTIONS)
+            .columns(60)
+            .rows(12)
+            .testUI(BaselineTest::createUI)
+            .positionTestUIBottomRowCentered()
+            .build()
+            .awaitAndCheck();
+    }
+
+    public static JFrame createUI() {
+        JFrame frame = new JFrame("BaselineTest");
+        frame.setLayout(new GridLayout(2, 1));
+
+        JTextPane prefPane = new JTextPane();
+        initJTextPane(prefPane);
+        frame.add(new JScrollPane(prefPane));
+
+        JTextPane variablePane = new JTextPane();
+        variablePane.setEditorKit(new CustomEditorKit());
+        initJTextPane(variablePane);
+        frame.add(new JScrollPane(variablePane));
+        frame.setSize(800, 400);
+        return frame;
+    }
+
+    static void initJTextPane(JTextPane tp) {
+
+        try {
+            Document doc = tp.getDocument();
+
+            doc.insertString(0, " Default Size Text ", null);
+            tp.setCaretPosition(doc.getLength());
+            tp.insertComponent(new PaintLabel(0.0f));
+            tp.insertComponent(new PaintLabel(0.2f));
+            tp.insertComponent(new PaintLabel(0.5f));
+            tp.insertComponent(new PaintLabel(0.7f));
+            tp.insertComponent(new PaintLabel(1.0f));
+            SimpleAttributeSet set = new SimpleAttributeSet();
+            StyleConstants.setFontSize(set, 20);
+            tp.setCaretPosition(doc.getLength());
+            doc.insertString(doc.getLength(), " Large Size Text ", set);
+        } catch (BadLocationException ble) {
+        }
+    }
+
+}
+
+class PaintLabel extends JLabel {
```

Comment:
Could you please move all the supporting classes `PaintLabel`, `CustomComponentView`, `CustomEditorKit` into the main test class `BaselineTest` as static nested classes.

This greatly simplifies working with test in IDE, there'll be no duplicate classes.

Reply:
I guess I can but I don't know where you are seeing duplicate classes.

Reply:
Now that we use a flat layout for tests, marking a folder as a test source folder may result in duplicate classes if any of the tests in the current folder or another folder with tests has classes with the same name. It makes running tests from IDE impossible because the test classes fail to compile.

This issue often occurs with `Sysout` standard test machinery (that we're slowly getting rid of by replacing with `PassFailJFrame`) and other cases…

The chance of getting such conflicts / duplicate class names is significantly reduced if all the supporting classes are nested inside the main test class.